controllers/virtualmachinesnapshot/virtualmachinesnapshot_controller.go

```diff
@@ -169,6 +170,8 @@ func (r *Reconciler) ReconcileNormal(ctx *pkgctx.VirtualMachineSnapshotContext)
 		return ctrl.Result{}, errVMRefNil
 	}
 
+	ensureCSIVolumeSyncAnnotation(vmSnapshot)
```

Comment:
Can you move the conditions.IsTrue check here instead of the called method?  That would make it clear to the reader that we only evaluate the annotation if the snapshot is ready.